File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03748/p03748_num5_parsed.py
# Program to find the count of different possible sequences of colors of bricks
# The count is found modulo 10^9+7

def count_sequences(N, M):
    MOD = 10**9 + 7

    # Calculate factorials up to N+M and their modular inverses
    factorial = [1] * (N + M + 1)
    inverse_factorial = [1] * (N + M + 1)
    for i in range(1, N + M + 1):
        factorial[i] = (factorial[i - 1] * i) % MOD
        inverse_factorial[i] = pow(factorial[i], MOD - 2, MOD)

    # Calculate the count of sequences
    count = 0
    for i in range(M + 1):
        # Calculate the count for each possible number of red bricks
        red_bricks = N + i - 1
        blue_bricks = i
        combinations = (factorial[red_bricks] * inverse_factorial[N - 1] % MOD) * inverse_factorial[blue_bricks] % MOD
        count = (count + combinations) % MOD

    return count

# Main program
N, M = map(int, input().split())
result = count_sequences(N, M)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03748/p03748_num5_parsed.cpp
#include <iostream>
#include <vector>

const int MOD = 1000000007;

int main() {
    // Read input
    int N, M;
    std::cin >> N >> M;

    // Initialize dp table
    std::vector<std::vector<long long>> dp(M + 1, std::vector<long long>(N + 1, 0));

    // Base case
    dp[0][0] = 1;

    // Dynamic programming
    for (int i = 1; i <= M; i++) {
        for (int j = 0; j <= N; j++) {
            dp[i][j] = (dp[i][j] + dp[i - 1][j] + dp[i - 1][j]) % MOD; // Same color bricks
            if (j > 0) {
                dp[i][j] = (dp[i][j] + dp[i - 1][j - 1]) % MOD; // Different color bricks
            }
            if (j < N) {
                dp[i][j] = (dp[i][j] + dp[i - 1][j + 1]) % MOD; // Different color bricks
            }
        }
    }

    // Calculate the count of possible sequences
    long long count = 0;
    for (int j = 0; j <= N; j++) {
        count = (count + dp[M][j]) % MOD;
    }

    // Print the result
    std::cout << count << std::endl;

    return 0;
}